cli/build.gradle.kts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+
+plugins {
+  `java-library`
+  `maven-publish`
+  alias(libs.plugins.nessie.run)
+  `build-conventions`
+}
+
+java.sourceCompatibility = JavaVersion.VERSION_1_8
+
+applyShadowJar()
+
+dependencies {
+  implementation(project(":iceberg-catalog-migrator-api"))
+  implementation(libs.slf4j)
+  runtimeOnly(libs.logback.classic)
+  implementation(libs.picocli)
+  implementation(libs.iceberg.spark.runtime)
+  implementation(libs.iceberg.dell)
+  implementation(libs.hadoop.aws) { exclude("com.amazonaws", "aws-java-sdk-bundle") }
+  implementation(libs.hadoop.common)
```

Comment:
There are still a lot of unneeded dependencies pulled in via hadoop-common - I thought that's been fixed?

Reply:
I just excluded what is excluded in iceberg repo now. I am not really sure whether catalogs really need these non-excluded dependencies. Only way to figure out is by testing all the catalogs which is a non-trivial task as I need to configure glue, ecs, jdbc etc. I will check it .

Reply:
Runtime dependencies like `hadoop-aws` and `hive-metastore` depends on `hadoop-common`. As I am unsure what and all classes from `hadoop-common` is used in those two runtime dependencies, I didn't exclude most of the transitive dependencies from `hadoop-common`.